Composer/packages/client/src/pages/publish/Publish.tsx

```diff
@@ -426,7 +427,13 @@ const Publish: React.FC<RouteComponentProps<{ projectId: string; targetName?: st
       <Pivot
         selectedKey={activeTab}
         styles={{ root: { marginLeft: 12 } }}
-        onLinkClick={(link) => setActiveTab(link?.props?.itemKey || '')}
+        onLinkClick={(link) => {
+          setActiveTab(link?.props?.itemKey || '');
+          if (link?.props.itemKey) {
+            setActiveTab(link.props.itemKey);
+            navigateTo(`/bot/${projectId}/publish/all/#${link.props.itemKey}`);
```

Comment:
Is calling navigateTo the right way to update the URL? this doesn't cause extra navigation to happen?

Reply:
It does not lead to a page refresh but does rerender the component